### PR TITLE
Multiple improvement, mainly related to error-handling.

### PR DESCRIPTION
**Pull request by aamirbutt**
This makes a lot of improvements. 

1. The results of api calls are displayed better.
2. Linter issues have been fixed.
3. Moved to Exceptions, and handling those at a single place now.
4. Used the created corpus id in the first call to do operations on. This avoids creating new corpora everytime the example is run.
5. Added NewtonSoft JSON library to parse the results (in case of Rest examples)
6. Upgraded to dotnet 7.0 (latest and greatest)

@tallatshafaat I have added you and Shane as reviewers. I don't know if there is anybody else who can review C# code. Please suggest.

### CODE REVIEW

File: language-examples/csharp/grpc/GrpcApiKeyQueries.cs [hunk 1]
```diff
@@ -85,11 +93,29 @@ private static void Query(long customerId, long corpusId, String query, String s
...
+                    if (status.Code != Com.Vectara.StatusCode.Ok)
+                    {
+                        Console.Error.WriteLine(string.Format("Failure status on query: {0}", status.StatusDetail));
+                    }
+                }
+                foreach (var responseSet in result.ResponseSet)
+                {
+                    foreach (var status in responseSet.Status)
+                    {
+                        if (status.Code != Com.Vectara.StatusCode.Ok)
```
Comment (tallatshafaat) [inline language-examples/csharp/grpc/GrpcApiKeyQueries.cs:107]: can `status` be null?

File: language-examples/csharp/grpc/GrpcBasicExample.cs [hunk 2]
```diff
@@ -129,12 +134,18 @@ private static String Index(long customerId, long corpusId, String indexingEndpo
                 });
 
                 var result = indexingClient.Index(request);
-                Console.WriteLine(result.ToString());
+                if (result.Status.Code == Com.Vectara.StatusCode.Ok) {
+                    Console.WriteLine("Document indexed successfully.");
+                } else if (result.Status.Code == Com.Vectara.StatusCode.AlreadyExists) {
+                    Console.WriteLine("Document was previously indexed.");
```
Comment (tallatshafaat) [inline language-examples/csharp/grpc/GrpcBasicExample.cs:140]: nit: perhaps add that upserts are not currently supported (or just append `result.Status.StatusDetail` to the log message).

File: language-examples/csharp/rest/RestApiKeyQueries.cs [hunk 3]
```diff
@@ -72,16 +83,37 @@ private static void Query(long customerId, long corpusId, String query, String s
...
                     HttpResponseMessage response = client.Send(request);
-                    String result = response.Content.ReadAsStringAsync().Result;
+                    string result = response.Content.ReadAsStringAsync().Result;
+                    JObject resultObj = JObject.Parse(result);
+                    JToken? responseSetArray = resultObj["responseSet"];
+                    if (responseSetArray == null)
+                    {
+                        throw new Exception("No results found");
+                    }
+                    foreach (var responseSet in responseSetArray)
```
Comment (tallatshafaat) [inline language-examples/csharp/rest/RestApiKeyQueries.cs:93]: Don't you need to check the status object here too inside responseset like you did in the grpc example?

File: language-examples/csharp/rest/RestQueryData.cs [hunk 4]
```diff
@@ -48,14 +51,33 @@ public static void Query(long customerId, long corpusId, String query, String se
...
+                {
+                    throw new Exception("No results found");
+                }
+                foreach (var responseSet in responseSetArray)
+                {
+                    JObject responseSetObj = JObject.Parse(responseSet.ToString());
+                    JToken? documents = responseSetObj["document"];
 
-                Console.WriteLine(result);
+                    foreach (JToken docSection in responseSetObj["response"])
```
Comment (tallatshafaat) [inline language-examples/csharp/rest/RestQueryData.cs:66]: Check status inside response?

**Comment by tallatshafaat** [comment]
Thanks for the improvements Aamir! Back in the day (2006?), I did quite a bit of C# coding :)
This looks good. I've left some minor comments.

**Review by tallatshafaat — APPROVED** [review]

File: language-examples/csharp/grpc/GrpcApiKeyQueries.cs (hunk 1 above)
Comment (aamirbutt) [reply language-examples/csharp/grpc/GrpcApiKeyQueries.cs:107]: Added a check.

File: language-examples/csharp/rest/RestApiKeyQueries.cs (hunk 3 above)
Comment (aamirbutt) [reply language-examples/csharp/rest/RestApiKeyQueries.cs:93]: Done.

File: language-examples/csharp/grpc/GrpcBasicExample.cs (hunk 2 above)
Comment (aamirbutt) [reply language-examples/csharp/grpc/GrpcBasicExample.cs:140]: Appended.

File: language-examples/csharp/rest/RestQueryData.cs (hunk 4 above)
Comment (aamirbutt) [reply language-examples/csharp/rest/RestQueryData.cs:66]: Done.

**Pull request merged**